yellowbrick/text/postag.py

```diff
@@ -114,251 +115,381 @@ def __init__(
         X : list or generator
             Should be provided as a list of documents or a generator
             that yields a list of documents that contain a list of
             sentences that contain (token, tag) tuples.
-
         y : ndarray or Series of length n
             An optional array of target values that are ignored by the
             visualizer.
-
         kwargs : dict
             Pass generic arguments to the drawing method
-
         Returns
         -------
         self : instance
             Returns the instance of the transformer/visualizer
         """
-        # TODO: add support for other tagsets?
+        if self.stack and y is None:
+            raise YellowbrickValueError("Specify y for stack=True")
+        
+        self.labels_=np.unique(y)
+        
         if self.tagset == "penn_treebank":
             self.pos_tag_counts_ = self._penn_tag_map()
-            self._handle_treebank(X)
-
+            self._handle_treebank(X,y)
         elif self.tagset == "universal":
             self.pos_tag_counts_ = self._uni_tag_map()
-            self._handle_universal(X)
-
+            self._handle_universal(X,y)
         self.draw()
         return self
 
     def _penn_tag_map(self):
         """
         Returns a Penn Treebank part-of-speech tag map.
         """
-        return {
-            "noun": 0,
-            "verb": 0,
-            "adjective": 0,
-            "adverb": 0,
-            "preposition": 0,
-            "determiner": 0,
-            "pronoun": 0,
-            "conjunction": 0,
-            "infinitive": 0,
-            "wh- word": 0,
-            "modal": 0,
-            "possessive": 0,
-            "existential": 0,
-            "punctuation": 0,
-            "digit": 0,
-            "non-English": 0,
-            "interjection": 0,
-            "list": 0,
-            "symbol": 0,
-            "other": 0,
-        }
+        
+        default_dic =  {
+                            "noun": 0,
+                            "verb": 0,
+                            "adjective": 0,
+                            "adverb": 0,
+                            "preposition": 0,
+                            "determiner": 0,
+                            "pronoun": 0,
+                            "conjunction": 0,
+                            "infinitive": 0,
+                            "wh- word": 0,
+                            "modal": 0,
+                            "possessive": 0,
+                            "existential": 0,
+                            "punctuation": 0,
+                            "digit": 0,
+                            "non-English": 0,
+                            "interjection": 0,
+                            "list": 0,
+                            "symbol": 0,
+                            "other": 0,
+                        }
+        if self.stack:
+            penn_tag_map={}
+            for i in self.labels_:
+                penn_tag_map[i]=default_dic.copy()
```

Comment:
Good use of `copy()` - that's very important! 

There is no need to make this change - but one thing I would think about looking at what's going on in `_penn_tag_map` and `_uni_tag_map` now is that there is some duplicate code that didn't exist before. If we wanted to add another tagset, we would have to duplicate this again. 

The only thing that is really different between these two functions is the names in the dictionary. Having the two functions before ensured that we were not passing around the same mutable object (e.g. the same reason you used `copy` here). 

One goal of YB is to make sure that the code is understandable to new contributors. One way to refactor the code might have been to add constants with the list of tags, then create a helper function that was called from the internal tag constructor functions. This would have made it easy to add new tags and also make the code a bit more readable to others:

```python
PENN_TAGS = [
    "noun", "verb", "adjective", "adverb", "preposition", "determiner", "pronoun",
    "conjunction", "infinitive", "wh- word", "modal", "possessive", "existential",
    "punctuation", "digit", "non-English", "interjection", "list", "symbol", "other", 
]

UNIVERSAL_TAGS = [...]

class PosTagVisualizer(TextVisualizer):

    def self._penn_tag_map(self):
        """
        Returns ...
        """
        return self._make_tag_map(PENN_TAGS)

    def self._uni_tag_map(self):
        """
        Returns ...
        """
        return self._make_tag_map(UNIVERSAL_TAGS)

    def _make_tag_map(self, tagset):
        """
        Returns a map of the tagset to a counter unless stack=True then returns
        a map of labels to a map of tagset to counters. 
        """
        zeros = [0] * len(tagset) # ensures the dict only contains a zero counter per tag
        if self.stack:
            return {
                label: dict(zip(tagset, zeros))
                for label in self.labels_ 
            }
        return dict(zip(tagset, zeros))
```